modules/sdk/src/main/scala/org/tessellation/sdk/domain/cluster/programs/Joining.scala

```diff
@@ -178,6 +179,7 @@ sealed abstract class Joining[F[_]: Async: GenUUID: SecurityProvider: KryoSerial
       _ <- sessionStorage.getToken.flatMap(_.fold(SessionDoesNotExist.raiseError[F, Unit])(_ => Applicative[F].unit))
 
       registrationRequest = joinRequest.registrationRequest
+      _ = InvalidNodeState.raiseError[F, Unit].unlessA(registrationRequest.state =!= Ready)
```

Comment:
as written on slack, this check is duplicated. also, it should be `_ <-` to flatMap the error into the chain of operations